app/uk/gov/hmrc/rasapi/repository/RasFileRepository.scala

```diff
@@ -15,116 +15,115 @@
 
-class RasFilesRepository @Inject()(val mongoComponent: ReactiveMongoComponent,
+class RasFilesRepository @Inject()(val mongoComponent: MongoComponent,
                                    val appContext: AppContext)
-                                   (implicit val ec: ExecutionContext) extends ReactiveRepository[Chunks, BSONObjectID](
+                                   (implicit val ec: ExecutionContext) extends PlayMongoRepository[Chunks](
+  mongoComponent = mongoComponent,
   collectionName = "rasFileStore",
-  mongo = mongoComponent.mongoConnector.db,
-  domainFormat = Chunks.format
+  domainFormat = Chunks.format,
+  indexes = Seq(),
+  replaceIndexes = false
 ) with GridFsTTLIndexing {
 
   override lazy val expireAfterSeconds: Long = appContext.resultsExpriyTime
-  private val contentType =  "text/csv"
   override val log: Logger = Logger(getClass)
-
-  val gridFSG: GridFS[BSONSerializationPack.type] =
-    GridFS[BSONSerializationPack.type](mongoComponent.mongoConnector.db(), "resultsFiles")
-
-  addAllTTLs(gridFSG)
-
-  def generateFileToSave(fileId: String, contentType: String, envelopeId: String, userId: String): FileToSave[BSONSerializationPack.type, BSONValue] =
-    DefaultFileToSave(
-      filename = Some(fileId),
-      contentType = Some(contentType),
-      uploadDate = Some(DateTime.now().getMillis),
-      metadata = BSONDocument("envelopeId" -> envelopeId, "fileId" -> fileId, "userId" -> userId)
-    )
+  private val contentType =  "text/csv"
+  private val bucketName: String = "resultsFiles"
+  val gridFSG: GridFSBucket = GridFSBucket(mongoComponent.database, bucketName)
 
   def saveFile(userId:String, envelopeId: String, filePath: Path, fileId: String): Future[ResultsFile] = {
     log.info("[RasFileRepository][saveFile] Starting to save file")
 
-    gridFSG.writeFromInputStream(generateFileToSave(fileId, contentType, envelopeId, userId), new FileInputStream(filePath.toFile)).map{ res =>
-      log.warn(s"Saved File id is ${res.id} for userId ($userId)")
-      res
-    }.recover{ case ex: Throwable =>
-        log.error(s"error saving file -> $fileId for userId ($userId). Exception: ${ex.getMessage}")
-        throw new RuntimeException(s"failed to save file due to error" + ex.getMessage)
+    val observableToUploadFrom: Observable[ByteBuffer] = Observable(
+      Seq(ByteBuffer.wrap(Files.readAllBytes(filePath).array)
+    ))
+
+    val options: GridFSUploadOptions = new GridFSUploadOptions()
+      .metadata(Document(
+        "envelopeId" -> envelopeId,
+        "fileId" -> fileId,
+        "userId" -> userId,
+        "contentType" -> contentType))
+
+    gridFSG.uploadFromObservable(fileId, observableToUploadFrom, options).head().flatMap { res =>
+      log.warn(s"[RasFileRepository][saveFile] Saved file $fileId for user $userId")
+      checkAndEnsureTTL(mongoComponent.database, s"$bucketName.files")
+      gridFSG.find(Document("_id" -> res)).head()
```

Comment:
just to be sure: could this result in trouble? e.g. could an Observable actually fail to upload, yet this still go into the .flatMap block?

Reply:
I believe it returns `null` so will likely get to `find` and then fall over into the recover block.

Reply:
ah yes I forgot the .recover exists here, happy with this one then